### DISCUSSION TITLE
awaiting a got call seems to return the got object, not the result

### DISCUSSION THREAD

**Discussion by jmls** [sindresorhus/got#1628] [Q&A] [answered]
#### Describe the bug

 - Node.js version: 15.5
 - OS & version: Ubuntu 

#### Actual behavior
the returned value from a got call is wrong

#### Expected behavior
data to be returned

#### Code to reproduce
```
       const got = require('got');
       
       async someFunc() {
          const x = await got(
               `http://myserver/ping`,
               { method: 'get', retry: 0, responseType: 'json' }
           );

           console.log('ping', x);
        }
```

gives the result 

```
 ping <ref *1> PassThrough {
  _readableState: ReadableState {
    objectMode: false,
    highWaterMark: 16384,
    buffer: BufferList { head: null, tail: null, length: 0 },
    length: 0,
    pipes: [],
    flowing: false,
    ended: true,
    endEmitted: true,
    reading: false,
    constructed: true,
    sync: false,
    needReadable: false,
    emittedReadable: false,
    readableListening: true,
     [snip]
    retryCount: 0,
    rawBody: <Buffer 7b 7d>,
    body: {},
    [Symbol(kCapture)]: false,
    [Symbol(kCallback)]: null
  }
```

a curl call works

```
curl http://myserver/ping
{}
```

I have used this library loads of time (with the same node and got version) so I don't understand what is happening in this project

- [x ] I have read the documentation.
- [x ] I have tried my code with the latest version of Node.js and Got.

**Comment by szmarczak** [comment, marked as answer]
`resolveBodyOnly: true`

**Reply by jmls** [reply]
thanks - I've verified that this works in this instance. However, I have never used this option before, so I am wondering if there is any other thing thaaat could have the same effect, as all my other apps use got without this option specified and they work fine.

What am I missing here ?

**Reply by szmarczak** [reply]
`await got(...).json()`

**Reply by szmarczak** [reply]
https://github.com/sindresorhus/got#json-mode
